jenkins/default.groovy

```diff
@@ -0,0 +1,150 @@
+                utils.pushDockerhub('mozorg/bedrock_code')
+                utils.pushDockerhub('mozorg/bedrock_l10n', 'mozorg/bedrock')
+            }
+        } catch(err) {
+            utils.ircNotification([stage: 'Dockerhub Push Failed', status: 'warning'])
+        }
+    }
+}
+
+/**
+ * Do region first because deployment and testing should work like this:
+ * region1:
+ *   push image -> deploy app1 -> test app1 -> deploy app2 -> test app2
+ * region2:
+ *   push image -> deploy app1 -> test app1 -> deploy app2 -> test app2
+ *
+ * A failure at any step of the above should fail the entire job
+ */
+if ( config.apps ) {
+    milestone()
+    // default to usw only
+    def regions = config.regions ?: ['usw']
+    for (regionId in regions) {
+        def region = global_config.regions[regionId]
+        if (region.registry_port) {
+            def stageName = "Private Push: ${region.name}"
+            stage (stageName) {
+                try {
+                    utils.pushPrivateReg(region.registry_port, config.apps)
+                } catch(err) {
+                    utils.ircNotification([stage: stageName, status: 'failure'])
+                    throw err
+                }
+            }
+        }
+        for (appname in config.apps) {
+            appSuffix = config.app_name_suffix ?: ''
+            if ( config.demo ) {
+                appURL = utils.demoAppURL(appname, region)
+            } else {
+                appURL = "https://${appname}${appSuffix}.${region.name}.moz.works"
+            }
+            stageName = "Deploy ${appname}-${region.name}"
+            // ensure no deploy/test cycle happens in parallel for an app/region
+            lock (stageName) {
+                milestone()
+                stage (stageName) {
+                    withEnv(["DEIS_PROFILE=${region.deis_profile}",
+                             "DEIS_BIN=${region.deis_bin}",
+                             "DOCKER_PRIVATE_REPO=${appname}",
+                             "DEIS_APPLICATION=${appname}"]) {
+                        try {
+                            retry(3) {
+                                if (config.demo) {
+                                    withCredentials([[$class: 'StringBinding',
+                                                      credentialsId: 'SENTRY_DEMO_DSN',
+                                                      variable: 'SENTRY_DEMO_DSN']]) {
+                                        sh 'docker/bin/prep_demo.sh'
+                                    }
+                                }
+                                sh 'docker/bin/push2deis.sh'
+                            }
+                        } catch(err) {
+                            utils.ircNotification([stage: stageName, status: 'failure'])
+                            throw err
+                        }
+                    }
+                    utils.ircNotification([message: appURL, status: 'shipped'])
```

Comment:
I've slightly altered the notifications so that the 🚢 message means that the app has been deployed, and the 🎉 message below indicates that the integration tests have passed. Just FYI.